src/pages/form/advanced-form/components/TableForm.tsx

```diff
@@ -0,0 +1,297 @@
+import { Button, Divider, Input, Popconfirm, Table, message } from 'antd';
+import { PlusOutlined } from '@ant-design/icons';
+import React, { Fragment, PureComponent } from 'react';
+
+import isEqual from 'lodash.isequal';
+import styles from '../style.less';
+
+interface TableFormDateType {
+  key: string;
+  workId?: string;
+  name?: string;
+  department?: string;
+  isNew?: boolean;
+  editable?: boolean;
+}
+interface TableFormProps {
+  loading?: boolean;
+  value?: TableFormDateType[];
+  onChange?: (value: TableFormDateType[]) => void;
+}
+
+interface TableFormState {
+  loading?: boolean;
+  value?: TableFormDateType[];
+  data?: TableFormDateType[];
+}
+class TableForm extends PureComponent<TableFormProps, TableFormState> {
+  static getDerivedStateFromProps(nextProps: TableFormProps, preState: TableFormState) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [expected call-signature: 'getDerivedStateFromProps' to have a typedef](https://app.codacy.com/app/afc163/ant-design-pro/pullRequest?prid=4583897)